tests/components/dormakaba_dkey/test_config_flow.py

```diff
@@ -0,0 +1,251 @@
+"""Test the Dormakaba dKey config flow."""
+from unittest.mock import patch
+
+from bleak.exc import BleakError
+from py_dormakaba_dkey import errors as dkey_errors
+from py_dormakaba_dkey.models import AssociationData
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.dormakaba_dkey.const import DOMAIN
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult, FlowResultType
+
+from . import DKEY_DISCOVERY_INFO, NOT_DKEY_DISCOVERY_INFO
+
+from tests.common import MockConfigEntry
+
+
+async def test_user_step_success(hass: HomeAssistant) -> None:
```

Comment:
See the xiaomi ble config flow tests for all the cases that we normally test for bluetooth flows.

Eg:

https://github.com/home-assistant/core/blob/ecb1d93b2e4455158a14ee8015625e5a0bc25cb9/tests/components/xiaomi_ble/test_config_flow.py#L732

https://github.com/home-assistant/core/blob/ecb1d93b2e4455158a14ee8015625e5a0bc25cb9/tests/components/xiaomi_ble/test_config_flow.py#L801-L802

https://github.com/home-assistant/core/blob/ecb1d93b2e4455158a14ee8015625e5a0bc25cb9/tests/components/xiaomi_ble/test_config_flow.py#L820-L823